### ISSUE TITLE
--compact doesn’t work when --files is used

### ISSUE THREAD

**Issue by chris-morgan** [XAMPPRocky/tokei#812]
Tokei 12.1.2 on Arch Linux.

<details>

<summary>Normally, <code>--compact</code> works.</summary>

```console
$ tokei
===============================================================================
 Language            Files        Lines         Code     Comments       Blanks
===============================================================================
 Markdown                1           33            0           24            9
 TOML                    1           17           13            1            3
-------------------------------------------------------------------------------
 Rust                   40        10389         9381          244          764
 |- Markdown            29          701            0          686           15
 (Total)                          11090         9381          930          779
===============================================================================
 Total                  42        10439         9394          269          776
===============================================================================

$ tokei --compact
===============================================================================
 Language            Files        Lines         Code     Comments       Blanks
===============================================================================
 Markdown                1           33            0           24            9
 Rust                   40        10389         9381          244          764
 TOML                    1           17           13            1            3
===============================================================================
 Total                  42        10439         9394          269          776
===============================================================================
```

</details>

<details>

<summary>But when you combine it with <code>--files</code>, it stops having any effect.</summary>

```console
$ tokei --files
===================================================================================================================================================================================
 Language                                                                                                                Files        Lines         Code     Comments       Blanks
===================================================================================================================================================================================
 Markdown                                                                                                                    1           33            0           24            9
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ./README.md                                                                                                                             33            0           24            9
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 TOML                                                                                                                        1           17           13            1            3
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ./Cargo.toml                                                                                                                            17           13            1            3
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Rust                                                                                                                       40        10389         9381          244          764
 |- Markdown                                                                                                                29          701            0          686           15
 (Total)                                                                                                                              11090         9381          930          779
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ./src/util/mod.rs                                                                                                                        3            3            0            0
 ./src/util/atomic.rs                                                                                                                    13           10            0            3
 ./src/util/string.rs                                                                                                                   160          139            0           21
 ./src/doc/mod.rs                                                                                                                       208          189            0           19
 ./src/comp/image_cache/glyph.rs                                                                                                        361          317           15           29
 ./src/layout/font/fallback.rs                                                                                                           34           28            0            6
 ./src/gfx/gl.rs                                                                                                                         19           16            0            3
 ./build.rs                                                                                                                              15           12            0            3
 ./src/gfx/mod.rs                                                                                                                       250          212           11           27
 ./src/gfx/device.rs                                                                                                                    223          207            3           13
 ./src/main.rs                                                                                                                          592          570            5           17
-- ./src/comp/text.rs -------------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                 28           24            0            4
 |- Markdown                                                                                                                             16            0           16            0
 ./src/comp/text.rs                                                                                                                      28           24            0            4
-- ./src/comp/mod.rs --------------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                168          152            0           16
 |- Markdown                                                                                                                             12            0           12            0
 ./src/comp/mod.rs                                                                                                                      168          152            0           16
-- ./src/comp/image_cache/mod.rs --------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                144          122            0           22
 |- Markdown                                                                                                                             24            0           24            0
 ./src/comp/image_cache/mod.rs                                                                                                          144          122            0           22
-- ./src/layout/font/prelude.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                  3            2            0            1
 |- Markdown                                                                                                                              1            0            1            0
 ./src/layout/font/prelude.rs                                                                                                             3            2            0            1
-- ./src/comp/batch/mod.rs --------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                301          276            0           25
 |- Markdown                                                                                                                             14            0           14            0
 ./src/comp/batch/mod.rs                                                                                                                301          276            0           25
-- ./src/comp/image_cache/atlas.rs ------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                326          301            8           17
 |- Markdown                                                                                                                             13            0           13            0
 ./src/comp/image_cache/atlas.rs                                                                                                        326          301            8           17
-- ./src/layout/layout_data.rs ----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                279          245            1           33
 |- Markdown                                                                                                                             19            0           19            0
 ./src/layout/layout_data.rs                                                                                                            279          245            1           33
-- ./src/layout/span_style.rs -----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                127          113            0           14
 |- Markdown                                                                                                                             19            0           19            0
 ./src/layout/span_style.rs                                                                                                             127          113            0           14
-- ./src/layout/font/system.rs ----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                 33           25            0            8
 |- Markdown                                                                                                                              2            0            2            0
 ./src/layout/font/system.rs                                                                                                             33           25            0            8
-- ./src/util/fxhash.rs -----------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                296          230           11           55
 |- Markdown                                                                                                                             44            0           36            8
 ./src/util/fxhash.rs                                                                                                                   296          230           11           55
-- ./src/comp/image_cache/cache.rs ------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                446          398           26           22
 |- Markdown                                                                                                                             16            0           16            0
 ./src/comp/image_cache/cache.rs                                                                                                        446          398           26           22
-- ./src/layout/font/library.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                 38           33            0            5
 |- Markdown                                                                                                                              3            0            3            0
 ./src/layout/font/library.rs                                                                                                            38           33            0            5
-- ./src/layout/font/shared_data.rs -----------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                101           86            1           14
 |- Markdown                                                                                                                              9            0            9            0
 ./src/layout/font/shared_data.rs                                                                                                       101           86            1           14
-- ./src/layout/font/index_data.rs ------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                278          248           18           12
 |- Markdown                                                                                                                              2            0            2            0
 ./src/layout/font/index_data.rs                                                                                                        278          248           18           12
-- ./src/layout/font/mod.rs -------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                116           89            3           24
 |- Markdown                                                                                                                             18            0           18            0
 ./src/layout/font/mod.rs                                                                                                               116           89            3           24
-- ./src/comp/shaders.rs ----------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                134          133            0            1
 |- Markdown                                                                                                                              5            0            5            0
 ./src/comp/shaders.rs                                                                                                                  134          133            0            1
-- ./src/layout/mod.rs ------------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                 49           37            0           12
 |- Markdown                                                                                                                              6            0            6            0
 ./src/layout/mod.rs                                                                                                                     49           37            0           12
-- ./src/layout/layout.rs ---------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                613          545            5           63
 |- Markdown                                                                                                                             52            0           52            0
 ./src/layout/layout.rs                                                                                                                 613          545            5           63
-- ./src/layout/line_breaker.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                431          374           44           13
 |- Markdown                                                                                                                              8            0            8            0
 ./src/layout/line_breaker.rs                                                                                                           431          374           44           13
-- ./src/layout/font/types.rs -----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                132          111            0           21
 |- Markdown                                                                                                                             10            0           10            0
 ./src/layout/font/types.rs                                                                                                             132          111            0           21
-- ./src/layout/font/family.rs ----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                134          120            0           14
 |- Markdown                                                                                                                              5            0            5            0
 ./src/layout/font/family.rs                                                                                                            134          120            0           14
-- ./src/layout/bidi.rs -----------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                826          771           20           35
 |- Markdown                                                                                                                             12            0           12            0
 ./src/layout/bidi.rs                                                                                                                   826          771           20           35
-- ./src/layout/nav.rs ------------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                696          609           45           42
 |- Markdown                                                                                                                             45            0           45            0
 ./src/layout/nav.rs                                                                                                                    696          609           45           42
-- ./src/layout/font/context.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                539          499            6           34
 |- Markdown                                                                                                                             49            0           48            1
 ./src/layout/font/context.rs                                                                                                           539          499            6           34
-- ./src/layout/builder_data.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                324          308            0           16
 |- Markdown                                                                                                                             60            0           60            0
 ./src/layout/builder_data.rs                                                                                                           324          308            0           16
-- ./src/layout/builder.rs --------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                594          567            7           20
 |- Markdown                                                                                                                             11            0           11            0
 ./src/layout/builder.rs                                                                                                                594          567            7           20
-- ./src/layout/font/builder.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                458          430            2           26
 |- Markdown                                                                                                                             18            0           18            0
 ./src/layout/font/builder.rs                                                                                                           458          430            2           26
-- ./src/layout/font/index.rs -----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                444          401            8           35
 |- Markdown                                                                                                                             24            0           24            0
 ./src/layout/font/index.rs                                                                                                             444          401            8           35
-- ./src/comp/color.rs ------------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                453          429            5           19
 |- Markdown                                                                                                                            184            0          178            6
 ./src/comp/color.rs                                                                                                                    453          429            5           19
===================================================================================================================================================================================
 Total                                                                                                                      42        10439         9394          269          776
===================================================================================================================================================================================

$ tokei --files --compact
===================================================================================================================================================================================
 Language                                                                                                                Files        Lines         Code     Comments       Blanks
===================================================================================================================================================================================
 Markdown                                                                                                                    1           33            0           24            9
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ./README.md                                                                                                                             33            0           24            9
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Rust                                                                                                                       40        10389         9381          244          764
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ./src/util/mod.rs                                                                                                                        3            3            0            0
 ./src/util/atomic.rs                                                                                                                    13           10            0            3
 ./src/util/string.rs                                                                                                                   160          139            0           21
 ./src/doc/mod.rs                                                                                                                       208          189            0           19
 ./src/comp/image_cache/glyph.rs                                                                                                        361          317           15           29
 ./src/layout/font/fallback.rs                                                                                                           34           28            0            6
 ./src/gfx/gl.rs                                                                                                                         19           16            0            3
 ./build.rs                                                                                                                              15           12            0            3
 ./src/gfx/device.rs                                                                                                                    223          207            3           13
 ./src/gfx/mod.rs                                                                                                                       250          212           11           27
 ./src/main.rs                                                                                                                          592          570            5           17
-- ./src/comp/text.rs -------------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                 28           24            0            4
 |- Markdown                                                                                                                             16            0           16            0
 ./src/comp/text.rs                                                                                                                      28           24            0            4
-- ./src/comp/mod.rs --------------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                168          152            0           16
 |- Markdown                                                                                                                             12            0           12            0
 ./src/comp/mod.rs                                                                                                                      168          152            0           16
-- ./src/comp/image_cache/mod.rs --------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                144          122            0           22
 |- Markdown                                                                                                                             24            0           24            0
 ./src/comp/image_cache/mod.rs                                                                                                          144          122            0           22
-- ./src/layout/layout_data.rs ----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                279          245            1           33
 |- Markdown                                                                                                                             19            0           19            0
 ./src/layout/layout_data.rs                                                                                                            279          245            1           33
-- ./src/layout/font/prelude.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                  3            2            0            1
 |- Markdown                                                                                                                              1            0            1            0
 ./src/layout/font/prelude.rs                                                                                                             3            2            0            1
-- ./src/comp/batch/mod.rs --------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                301          276            0           25
 |- Markdown                                                                                                                             14            0           14            0
 ./src/comp/batch/mod.rs                                                                                                                301          276            0           25
-- ./src/util/fxhash.rs -----------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                296          230           11           55
 |- Markdown                                                                                                                             44            0           36            8
 ./src/util/fxhash.rs                                                                                                                   296          230           11           55
-- ./src/layout/span_style.rs -----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                127          113            0           14
 |- Markdown                                                                                                                             19            0           19            0
 ./src/layout/span_style.rs                                                                                                             127          113            0           14
-- ./src/comp/image_cache/atlas.rs ------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                326          301            8           17
 |- Markdown                                                                                                                             13            0           13            0
 ./src/comp/image_cache/atlas.rs                                                                                                        326          301            8           17
-- ./src/layout/font/system.rs ----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                 33           25            0            8
 |- Markdown                                                                                                                              2            0            2            0
 ./src/layout/font/system.rs                                                                                                             33           25            0            8
-- ./src/layout/font/library.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                 38           33            0            5
 |- Markdown                                                                                                                              3            0            3            0
 ./src/layout/font/library.rs                                                                                                            38           33            0            5
-- ./src/layout/font/shared_data.rs -----------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                101           86            1           14
 |- Markdown                                                                                                                              9            0            9            0
 ./src/layout/font/shared_data.rs                                                                                                       101           86            1           14
-- ./src/comp/image_cache/cache.rs ------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                446          398           26           22
 |- Markdown                                                                                                                             16            0           16            0
 ./src/comp/image_cache/cache.rs                                                                                                        446          398           26           22
-- ./src/layout/mod.rs ------------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                 49           37            0           12
 |- Markdown                                                                                                                              6            0            6            0
 ./src/layout/mod.rs                                                                                                                     49           37            0           12
-- ./src/layout/line_breaker.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                431          374           44           13
 |- Markdown                                                                                                                              8            0            8            0
 ./src/layout/line_breaker.rs                                                                                                           431          374           44           13
-- ./src/layout/layout.rs ---------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                613          545            5           63
 |- Markdown                                                                                                                             52            0           52            0
 ./src/layout/layout.rs                                                                                                                 613          545            5           63
-- ./src/layout/font/index_data.rs ------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                278          248           18           12
 |- Markdown                                                                                                                              2            0            2            0
 ./src/layout/font/index_data.rs                                                                                                        278          248           18           12
-- ./src/layout/font/mod.rs -------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                116           89            3           24
 |- Markdown                                                                                                                             18            0           18            0
 ./src/layout/font/mod.rs                                                                                                               116           89            3           24
-- ./src/layout/font/types.rs -----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                132          111            0           21
 |- Markdown                                                                                                                             10            0           10            0
 ./src/layout/font/types.rs                                                                                                             132          111            0           21
-- ./src/layout/font/family.rs ----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                134          120            0           14
 |- Markdown                                                                                                                              5            0            5            0
 ./src/layout/font/family.rs                                                                                                            134          120            0           14
-- ./src/comp/shaders.rs ----------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                134          133            0            1
 |- Markdown                                                                                                                              5            0            5            0
 ./src/comp/shaders.rs                                                                                                                  134          133            0            1
-- ./src/layout/nav.rs ------------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                696          609           45           42
 |- Markdown                                                                                                                             45            0           45            0
 ./src/layout/nav.rs                                                                                                                    696          609           45           42
-- ./src/layout/bidi.rs -----------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                826          771           20           35
 |- Markdown                                                                                                                             12            0           12            0
 ./src/layout/bidi.rs                                                                                                                   826          771           20           35
-- ./src/layout/font/context.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                539          499            6           34
 |- Markdown                                                                                                                             49            0           48            1
 ./src/layout/font/context.rs                                                                                                           539          499            6           34
-- ./src/layout/builder.rs --------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                594          567            7           20
 |- Markdown                                                                                                                             11            0           11            0
 ./src/layout/builder.rs                                                                                                                594          567            7           20
-- ./src/layout/builder_data.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                324          308            0           16
 |- Markdown                                                                                                                             60            0           60            0
 ./src/layout/builder_data.rs                                                                                                           324          308            0           16
-- ./src/layout/font/builder.rs ---------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                458          430            2           26
 |- Markdown                                                                                                                             18            0           18            0
 ./src/layout/font/builder.rs                                                                                                           458          430            2           26
-- ./src/layout/font/index.rs -----------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                444          401            8           35
 |- Markdown                                                                                                                             24            0           24            0
 ./src/layout/font/index.rs                                                                                                             444          401            8           35
-- ./src/comp/color.rs ------------------------------------------------------------------------------------------------------------------------------------------------------------
 |- Rust                                                                                                                                453          429            5           19
 |- Markdown                                                                                                                            184            0          178            6
 ./src/comp/color.rs                                                                                                                    453          429            5           19
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 TOML                                                                                                                        1           17           13            1            3
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 ./Cargo.toml                                                                                                                            17           13            1            3
===================================================================================================================================================================================
 Total                                                                                                                      42        10439         9394          269          776
===================================================================================================================================================================================
```

(Example repository used was [swash_demo](https://github.com/dfrg/swash_demo).)

</details>

**Comment by XAMPPRocky** [comment]
Thank you for your issue!

**Comment by toyboot4e** [comment]
It's already fixed in #722 and works on HEAD!

**Comment by nicoburns** [comment]
Has this regressed? I'm getting markdown lines counts for Rust files with `tokei --compact --files` with tokei v12.1.2
